lib/ansible/modules/windows/win_mapped_drive.ps1

```diff
@@ -3,117 +3,347 @@
 # Copyright: (c) 2017, Ansible Project
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-#Requires -Module Ansible.ModuleUtils.Legacy
+#AnsibleRequires -CSharpUtil Ansible.Basic
+#Requires -Module Ansible.ModuleUtils.AddType
 
-$ErrorActionPreference = 'Stop'
+$spec = @{
+    options = @{
+        letter = @{ type = "str"; required = $true }
+        path = @{ type = "path"; }
+        state = @{ type = "str"; default = "present"; choices = "absent", "present" }
```

Comment:
Just a tiny tidbit, I am always confused of those implicit lists, probably because in Python comma is a list-separator, and it is easy to confuse the entries from the dictionary with those from the 'inline' list.